packages/visual-editor/src/store/tree.ts

```diff
@@ -46,6 +56,38 @@ export const useTreeStore = create<TreeStore>((set, get) => ({
     },
   },
   breakpoints: [],
+
+  updateNodesByUpdatedEntity: (entityId: string) => {
+    set(
+      produce((draftState: TreeStore) => {
+        treeVisit(draftState.tree.root, (node) => {
+          if (isAssemblyNode(node) && node.data.blockId === entityId) {
+            // Cannot use `structuredClone()` as node is probably a Proxy object with weird references
+            updateNode(node.data.id, cloneDeepAsPOJO(node), draftState.tree.root);
+            return;
+          }
+          const dataSourceIds = Object.values(node.data.dataSource).map((link) => link.sys.id);
+          if (dataSourceIds.includes(entityId)) {
+            // Cannot use `structuredClone()` as node is probably a Proxy object with weird references
+            updateNode(node.data.id, cloneDeepAsPOJO(node), draftState.tree.root);
+          }
+        });
+      })
+    );
+  },
+
+  /**
+   * Force updates entire tree. Usually shouldn't be used as updateTree()
```

Comment:
As this shouldn't be used in production, I would like to claim this clear at the front:
```suggestion
   * WARNING! Please note that this is for debugging purposes only:
   * Force updates entire tree. Usually shouldn't be used as updateTree()
```